impeller/geometry/gradient.cc

```diff
@@ -86,16 +89,90 @@ GradientData CreateGradientBuffer(const std::vector<Color>& colors,
         auto t = (scaled_i - previous_stop) / (next_stop - previous_stop);
         auto mixed_color = Color::lerp(previous_color, next_color, t);
 
-        AppendColor(mixed_color, &color_stop_channels);
+        AppendColor(mixed_color, &data);
       }
     }
     // The last index is always equal to the last color, exactly.
-    AppendColor(colors.back(), &color_stop_channels);
+    AppendColor(colors.back(), &data);
   }
-  return GradientData{
-      .color_bytes = std::move(color_stop_channels),
-      .texture_size = texture_size,
-  };
+  return data;
+}
+
+std::optional<std::vector<Color>> CreateGradientColors(
+    const std::vector<Color>& colors,
+    const std::vector<Scalar>& stops) {
+  FML_DCHECK(stops.size() == colors.size());
+
+  if (stops.size() == 2) {
+    // Use original buffer.
+    return std::nullopt;
+  }
+
+  uint32_t color_count;
+  auto minimum_delta = 1.0;
+  for (size_t i = 1; i < stops.size(); i++) {
+    auto value = stops[i] - stops[i - 1];
+    // Smaller than kEhCloseEnough
+    if (value < 0.0001) {
```

Comment:
Curious, why not just use the constant instead of hardcoding the value?

Reply:
context was sort of https://github.com/flutter/engine/pull/36218

TLDR, if you have really close stops you need a smaller number

Reply:
in the future, we'll need to stop computing the gradients on the CPU for certain stop values